### PR TITLE
Added Heston and CEV models

### PR DESCRIPTION
**Pull request by NirmaySriharsha**

### CODE REVIEW

File: mbt_gym/stochastic_processes/midprice_models.py [hunk 1]
```diff
@@ -290,3 +290,82 @@ def update(self, arrivals: np.ndarray, fills: np.ndarray, actions: np.ndarray) -
...
+            initial_state = np.array([[initial_price, initial_variance]]),
+            num_trajectories = num_trajectories,
+            seed = seed,
+        )
+    
+    def update(self, arrivals:np.ndarray,fills:np.ndarray, actions:np.ndarray)->np.ndarray:
+        weiner_means = np.array([0, 0])
+        weiner_corr = np.array([[1, self.weiner_correlation], [self.weiner_correlation,1]])
+        weiners = np.random.multivariate_normal(weiner_means, cov = weiner_corr, size=self.num_trajectories)
+        self.current_state[:,0] = self.current_state[:,0]*np.exp((self.drift - self.current_state[:,1]/2)*self.step_size
```
Comment (JJJerome) [inline mbt_gym/stochastic_processes/midprice_models.py:330]: I think that updates should be written using the Euler scheme of the process. Perhaps something along the lines of: https://www.quantstart.com/articles/Heston-Stochastic-Volatility-Model-with-Euler-Discretisation-in-C/ 

This is for two reasons:
(1) I am not 100% convinced that you can write the Heston in this explicit form. Have you got a reference fo how you coded it up?
(2) By removing the (fairly costly) exponential, simulation should be sped up a bit. 

P.S. I noted that the GBM implementation also uses the exponential, and I will change that.

File: mbt_gym/stochastic_processes/midprice_models.py [hunk 2]
```diff
@@ -290,3 +290,82 @@ def update(self, arrivals: np.ndarray, fills: np.ndarray, actions: np.ndarray) -
...
+                                + np.sqrt(self.current_state[:,1])*np.sqrt(self.step_size)*weiners[:,0])
+        self.current_state[:,1] = np.abs(self.current_state[:,1] + self.volatility_mean_reversion*(self.long_variance-self.current_state[:, 1])*self.step_size + self.volatility_volatility*np.sqrt(self.current_state[:, 1])*np.sqrt(self.step_size)*weiners[:,1])
+
+    def _get_max_value(self, initial_price, terminal_time):
+        return initial_price + 4 * self.long_variance * terminal_time
+
+class ConstantElasticityofVarianceMidpriceModel(MidpriceModel):
```
Comment (JJJerome) [inline mbt_gym/stochastic_processes/midprice_models.py:337]: Capital O for of

**Review by JJJerome — CHANGES_REQUESTED** [review]
This looks good!

The only real change I would suggest is to use the Euler scheme for the asset price in the Heston model. Also some nitpicking.

**Comment by NirmaySriharsha** [comment]
Added a Euler scheme. I actually don't exactly remember what's going on with self.max_depth in TradingEnvironment. I remember when I ran check_all_imports.py the TradingEnvironment.py file was throwing up errors because the default value for max_depth is None and that particular section of code was interpreting it as a float. I think what I did was just a temporary workaround to get past some meaningless type errors. Discard those changes if they're not necessary.

**Review by JJJerome — DISMISSED** [review]

**Review by leandro-sbetancourt — APPROVED** [review]

**Comment by NirmaySriharsha** [comment]
I don't have write access permission to merge.

**Review by JJJerome — APPROVED** [review]
LGTM! Will merge.

**Pull request merged**